cli/lsp/tsc.rs

```diff
@@ -4640,6 +4642,7 @@ fn op_resolve_inner(
       })
     })
     .collect();
+  NodeResolutionThreadLocalCache::clear();
```

Comment:
We should improve this to be smarter. Right now I just have it clearing after each op_resolve.

Reply:
Could we clear it after a request is completed instead? I did a similar thing in https://github.com/denoland/deno/pull/27831

Reply:
Oh, actually I think we can just clear when the dependents and config are recreated. I will move this there.

Reply:
Ah, seems tricky. Yeah I'm going to switch to clear after the request is completed.